zio-http/src/main/scala/zio/http/Body.scala

```diff
@@ -126,10 +126,20 @@ trait Body { self =>
    */
   def isEmpty: Boolean
 
-  private[zio] def mediaType: Option[MediaType]
+  /**
+    * Returns the media type for this Body 
+    */
+  def mediaType: Option[MediaType]
+
+  /**
+    * Updates the media type attached to this body, returning a new Body
+    * with the updated media type 
+    */
+  def mediaType(newMediaType: MediaType): Body
+
   private[zio] def boundary: Option[Boundary]
 
-  private[zio] def contentType(newMediaType: MediaType, newBoundary: Option[Boundary] = None): Body
+  def contentType(newMediaType: MediaType, newBoundary: Option[Boundary] = None): Body
```

Comment:
now we have mediaType, that takes a mediaType and contentType that takes a boundary as well. This feels odd to me. I think we should just have one name, and two overloaded methods, one with only media type and one with a non optional boundary.

Reply:
Yeah i agree it is a bit confusing. Will apply the suggestion